app/scripts/controllers/header.controller.js

```diff
@@ -20,7 +20,7 @@
         };
 
         $scope.toggleDarkTheme = function() {
-            var isUsingDarkTheme = localStorage.getItem('darkTheme') === 'true';
+            var isUsingDarkTheme = $scope.$parent.isUsingDarkTheme;
```

Comment:
I don't think that works. Imagine the following situation:

1. user arrives with `prefers-color-scheme = dark` 
2. this function gets run, and sets `darkTheme = true` in localStorage
2. user changes `prefers-color-scheme` value
4. the changes don't reflect in the website, since the localStorage was overwritten by the function, and now the website considers that the user only wants darkTheme.

When using the prefers-color-scheme emulation in Chrome, I can confirm that the website doesn't react to changes in this setting, as it sets it in the localStorage and never looks at this value again.